src/widget/wtracktableview.cpp

```diff
@@ -299,7 +300,7 @@ void WTrackTableView::loadTrackModel(QAbstractItemModel *model) {
     // restoring scrollBar position using model pointer as key
     // scrollbar positions with respect to different models are backed by map
 
-    m_pMenu = new WTrackMenu(this, m_pConfig, m_pTrackCollectionManager, WTrackMenu::Filter::All, trackModel);
+    m_pMenu = make_parented<WTrackMenu>(this, m_pConfig, m_pTrackCollectionManager, WTrackMenu::Filter::All, trackModel);
```

Comment:
I'm not sure this actually takes care of the memory leak because it is creating more `parented_ptr` objects during the lifetime of the Qt parent object. @uklotzde can you take a look?

Reply:
WTrackMenu object is wrapped in a QScopedPointer. Shouldn't this take care of deleting a dangling pointer? Again, what is the possibility of the loadTrackModel function being called more than once?

Reply:
QScopePointer? You are using parented_ptr here.

> what is the possibility of the loadTrackModel function being called more than once?

I'm not sure. Use your IDE to navigate the code and find out.

Reply:
> QScopedPointer? You are using parented_ptr here.
I might have confused it with some other pointer. But should we switch to QScopedPointer/unique_ptr or keep using parented_ptr?

Reply:
I'm not sure. I'm confused about the appropriate use cases of parented_ptr. @uklotzde or @daschuer can you help here?

Reply:
We have not yet reached a consensus on this @Be-ing.

Reply:
Yeah I'm not sure what the best smart pointer to use here. Waiting on a response from @uklotzde or @daschuer.

Reply:
parented_ptr has its [delete operator intentionally disabled](https://github.com/mixxxdj/mixxx/blob/master/src/util/parented_ptr.h#L62), but we need to delete the pointer here, so parented_ptr is not appropriate. I think we should use unique_ptr.

Reply:
QScopedPointer acts like a weak pointer. It may become `null` at any time if the referenced object is deleted. It is **not** intended for managing ownership!

Reply:
So what do you recommend here? Is unique_ptr the most appropriate? Or is there a better option?

Reply:
`this` is the parent and owner of `m_pMenu`. Therefore `parented_ptr` is the correct choice. This class is nothing more then a complex debug assertion, it does essentially nothing. It just checks in its destructor that the internal pointer is either `nullptr` or that it is still owned by someone.

Reply:
So do you recommend using `parented_ptr::get` and deleting the raw pointer? That seems to me like it somewhat defeats the purpose of parented_ptr.

Reply:
Oh, now I got it. All menu objects will live on until `this` finally gets deleted. In this case we need to delete the object manually.

```
if (m_pMenu) {
    m_pMenu->deleteLater();
}
m_pMenu = make_parented<WTrackMenu>(...);
```

Reply:
Since we assign a newly created object immediately after we don't need to invalidate the pointer manually by `m_pMenu = nullptr;`. Otherwise I would recommend this.

Reply:
The `loadTrackModel` slot is essentially called once during a session for each of its objects. But in order to accommodate the possibility of it being called more than once, we sure need to use a smart pointer or take care of explicitly deleting in case of re-initialization.
So are we good with this solution?
>```cpp
if (m_pMenu) {
    m_pMenu->deleteLater();
}
m_pMenu = make_parented<WTrackMenu>(...);
```